### PR TITLE
working cloud engineer permissions

### PR DESCRIPTION
**Pull request by HrisyToteva**
closes dfds/cloudplatform/issues/2832

# Additional Review Notes
Functionality to disable the cloud engineer role permissions

### CODE REVIEW

File: src/SelfService/Domain/Services/AuthorizationService.cs [hunk 1]
```diff
@@ -70,8 +73,17 @@ public async Task<bool> CanChange(PortalUser portalUser, KafkaTopic kafkaTopic)
 
     public bool CanViewDeletedCapabilities(PortalUser portalUser)
     {
+        Console.WriteLine();
```
Comment (andreasfrisch) [inline src/SelfService/Domain/Services/AuthorizationService.cs:76]: Is this needed anymore?

File: src/SelfService/Domain/Services/AuthorizationService.cs [hunk 2]
```diff
@@ -247,7 +319,20 @@ public async Task<bool> CanRetryCreatingMessageContract(PortalUser portalUser, M
         }
         var kafkaTopic = await _kafkaTopicRepository.Get(messageContract.KafkaTopicId);
         bool isMember = await _membershipQuery.HasActiveMembership(portalUser.Id, kafkaTopic.CapabilityId);
-        bool isCloudEngineer = portalUser.Roles.Any(role => role == UserRole.CloudEngineer);
+        bool isCloudEngineer = false;
+
+        if (portalUser.Roles.Any(role => role == UserRole.CloudEngineer))
```
Comment (andreasfrisch) [inline src/SelfService/Domain/Services/AuthorizationService.cs:324]: We have this particular code in multiple parts; Should we extract it and thus simplify these sections? :)

File: src/SelfService/Infrastructure/Api/UserImpersonation.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        context.Request.Headers.TryGetValue("x-selfservice-permissions", out StringValues permissions);
+
+        if (permissions.ToString().Equals("1"))
+        {
+            if (context.User.ToPortalUser().Roles.Count(x => x.ToString().Equals("CloudEngineer")) == 1)
+            {
+                context.Items["userPermissions"] = "1";
```
Comment (andreasfrisch) [inline src/SelfService/Infrastructure/Api/UserImpersonation.cs:18]: Should we introduce a new Enum instead?

File: src/SelfService/Domain/Services/AuthorizationService.cs [hunk 4]
```diff
@@ -70,8 +73,17 @@ public async Task<bool> CanChange(PortalUser portalUser, KafkaTopic kafkaTopic)
 
     public bool CanViewDeletedCapabilities(PortalUser portalUser)
     {
+        Console.WriteLine();
         if (portalUser.Roles.Any(role => role == UserRole.CloudEngineer))
         {
+            if (
+                _httpContextAccessor.HttpContext != null
+                && _httpContextAccessor.HttpContext.Items.ContainsKey("userPermissions")
```
Comment (andreasfrisch) [inline src/SelfService/Domain/Services/AuthorizationService.cs:81]: Should we care about the content of the userPermission key?

**Review by andreasfrisch — CHANGES_REQUESTED** [review]
I am really glad that we finally have this, but I think we should do things a slightly different way 👍 
Let's talk about this :)

**Review by SEQUOIIA — APPROVED** [review]

**Pull request merged**